From 1406456b64f6cd221a7b58234ce7813ad2688aab Mon Sep 17 00:00:00 2001
From: Alexey Puzhevich <alexey.puzhevich@intel.com>
Date: Tue, 13 May 2025 13:38:27 +0300
Subject: [PATCH] [SW-229368] palad: spi flash hexfile generation fix

Fix changes the current directory to stay in the
buildroot-external folder, as FLASH_BIN_FILE is
referenced relatively to the buildroot-external folder.

Change-Id: Ib3428d132c4169f398495e56e60b0828912be699
Signed-off-by: Alexey Puzhevich <alexey.puzhevich@intel.com>
---

diff --git a/scripts/make/hl-common/Makefile b/scripts/make/hl-common/Makefile
index d5b620b..b93e746 100644
--- a/scripts/make/hl-common/Makefile
+++ b/scripts/make/hl-common/Makefile
@@ -430,7 +430,5 @@
 %-bin-create:
 	BIN_DIR=output/${HL_DEV}/host/bin $(HABANA_SOFTWARE_STACK)/automation/habana_scripts/embedded/scripts/itb2spi.sh -s $(addprefix -i $(ITBS_DIR),$(HL_BIN_FITS)) -o $(FLASH_BIN_FILE) && \
 	source ${BR2_EXTERNAL}/board/habanalabs/common/image-utility.sh && \
-	pushd $(STAGE_DIR) && \
 	swap_file $(FLASH_BIN_FILE) $(FLASH_BIN_FILE).rev && \
-	hl_palad_flash_hex spi-flash $(FLASH_BIN_FILE).rev $(HEXS_DIR) && \
-	popd
+	hl_palad_flash_hex spi-flash $(FLASH_BIN_FILE).rev $(HEXS_DIR)
